AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConfigurationTests.swift

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSMobileClient
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreConfigurationTests: XCTestCase {
+
+    func testConfigureWithSameSchemaDoesNotDeleteDatabase() throws {
+        Amplify.reset()
+
+        let prevoisVersion = "123"
+
+        let saveSuccess = expectation(description: "Save was successful")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(version: prevoisVersion))
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure(AmplifyConfiguration(dataStore: nil))
+        } catch {
+            XCTFail("Failed to initialize Amplify with \(error)")
+        }
+
+        let post = Post(title: "title", content: "content", createdAt: .now())
+
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                saveSuccess.fulfill()
+            case .failure(let error):
+                XCTFail("Error saving post \(error)")
+            }
+        })
+
+        wait(for: [saveSuccess], timeout: TestCommonConstants.networkTimeout)
+
+        Amplify.reset()
+
+        let querySuccess = expectation(description: "Database remains")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(version: prevoisVersion))
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure(AmplifyConfiguration(dataStore: nil))
+        } catch {
+            XCTFail("Failed to initialize Amplify with \(error)")
+        }
+
+        // Here trying to query the saved data from above, because if the schema hasn't been changed that the database and saved data should persist.
+        Amplify.DataStore.query(Post.self, byId: post.id) { result in
+            switch result {
+            case .success(let postOptional):
+                querySuccess.fulfill()
```

Comment:
fulfill should come at the end or else the wait till be released and finish executing at the same time as the XCTAsserts u have below the fulfill

Reply:
I see. I thought it is going to be fulfill first and then check the retrieved data